v2/libs/keystore/keystore.ts

```diff
@@ -183,16 +225,52 @@ export class Keystore {
 			type: 'internal',
 			label,
 			// @TODO: consider an MAC?
-			privKey: hexlify(aesCtr.encrypt(getBytes(privateKey))),
+			privKey: hexlify(aesCtr.encrypt(aes.utils.hex.toBytes(privateKey))),
 			meta: null
 		})
 		await this.storage.set('keystoreKeys', keys)
 	}
 
 	async removeKey(id: string) {
 		const keys: [StoredKey] = await this.storage.get('keystoreKeys', [])
-		if (!keys.find(x => x.id === id)) throw new Error(`keystore: trying to remove key that does not exist ${id}}`)
-		this.storage.set('keystoreKeys', keys.filter(x => x.id !== id))
+		if (!keys.find((x) => x.id === id)) throw new Error(`keystore: trying to remove key that does not exist ${id}}`)
+		this.storage.set(
+			'keystoreKeys',
+			keys.filter((x) => x.id !== id)
+		)
+	}
+
+	async getSigner(keyId: string) {
+		const storedKey: StoredKey = (await this.storage.get('keystoreKeys', [])).find((x: StoredKey) => x.id === keyId)
+
+		if (!storedKey) throw new Error('keystore: key not found')
+		const { id, label, type, meta } = storedKey
+
+		const key = {
+			id,
+			label,
+			type,
+			meta,
+			isExternallyStored: type !== 'internal'
+		}
+
+		const signerInitializer = this.keystoreSigners[key.type]
+		if (!signerInitializer) throw new Error('keystore: unsupported singer')
+		if (key.type === 'internal' && !this.isUnlocked()) throw new Error('keystore: not unlocked')
+
+		if (key.type === 'internal') {
+			const encryptedBytes = getBytes(storedKey.privKey as string)
```

Comment:
you can put `if (!this.unlocked()) throw new Error` in this branch instead of having a separate check, it will clean up the logic